lib/models/LifecycleConfiguration.js

```diff
@@ -563,9 +563,11 @@ class LifecycleConfiguration {
             return undefined;
         }
         const { days, date, storageClass, rule } = params;
-        const invalidTransition = rule.Transition.find(transition => {
-            if (storageClass === transition.StorageClass[0]) {
-                return undefined;
+        let compareStorageClass;
+        const invalidTransition = rule.Transition.some(transition => {
+            compareStorageClass = transition.StorageClass[0]
```

Comment:
minor: you can declare `compareStorageClass` here as a scoped `const`, making it clear it's only valid in this block.

Reply:
nevermind, I did not see you use it in the error message

Reply:
I'm going to change it back to using `.find` (but keeping the boolean return values) to make this more obvious.